ts/src/index.ts

```diff
@@ -79,7 +78,20 @@ export async function initConfig(config: Config): Promise<ProfilerConfig> {
   const mergedConfig =
       extend(true, {}, defaultConfig, envSetConfig, envConfig, config);
 
-  if (!mergedConfig.zone || !mergedConfig.instance) {
+  if (!hasService(mergedConfig)) {
```

Comment:
Q: does this change make this a semver major change? That is, previously working code may no longer work if it relied on the service context being discovered from the metadata service?

Reply:
We've never checked metadata for the service, so this would not be semver major.